pennylane/gradients/finite_difference.py

```diff
@@ -177,6 +176,16 @@ def _processing_fn(results, shots, single_shot_batch_fn):
     return tuple(grads_tuple)
 
 
+def _finite_diff_stopping_condition(op) -> bool:
+    if not op.has_decomposition:

```

Comment:
I checked `tape.py::expand_tape` and think I understood that this is no change in behaviour. Want to ask nonetheless, this is unchanged?

Reply:
This basically matches the stopping condition for `expand_invalid_trainable`.  I add in the `if not op.has_decomposition` check to better match pre-existing behavior for `tape.expand`.  Fewer things to update this way.

Reply:
Thanks!